### PR TITLE
K2: add new CI steps

### PR DESCRIPTION
**Pull request by apolyakov**

### CODE REVIEW

File: tests/kphp_ci_pipeline_runner.py [hunk 1]
```diff
@@ -170,10 +170,18 @@ def parse_args():
         type=str,
         dest="cxx_name",
         default="g++",
-        choices=["g++", "clang++", "clang++-16"],
+        choices=["g++", "clang++", "clang++-16", "clang++-18", "clang++-19"],
```
Comment (mkornaukhov03) [inline tests/kphp_ci_pipeline_runner.py:173]: I think we should throw away clang16.
The current solution raises a question: why clang16 is here, but clang17 is not?
The last release of clang16 was more than a year ago, it's obsolete now, v18 and v19 are good I think

File: tests/kphp_ci_pipeline_runner.py [hunk 2]
```diff
@@ -281,6 +309,20 @@ def parse_args():
...
+        description="run k2-kphp tests with cxx={}".format(args.cxx_name),
+        cmd="KPHP_TESTS_POLYFILLS_REPO={kphp_polyfills_repo} "
+            "{kphp_runner} -j{jobs} --cxx-name {cxx_name} --k2-bin {k2_bin}".format(
+            jobs=n_cpu,
+            kphp_polyfills_repo=kphp_polyfills_repo,
+            kphp_runner=kphp_test_runner,
+            cxx_name=args.cxx_name,
+            k2_bin=args.k2_bin,
+        ),
+        skip=not args.k2_bin or args.steps and "k2-kphp-tests" not in args.steps,
```
Comment (mkornaukhov03) [inline tests/kphp_ci_pipeline_runner.py:323]: Too hard to parse, I would add some parentheses

**Review by mkornaukhov03 — CHANGES_REQUESTED** [review]

File: tests/kphp_ci_pipeline_runner.py (hunk 1 above)
Comment (apolyakov) [reply tests/kphp_ci_pipeline_runner.py:173]: `clang16` is used in current `KPHP` pipelines. Removing it also requires modifying the pipelines. It's removal definitely should be a distinct step.

File: tests/kphp_ci_pipeline_runner.py (hunk 2 above)
Comment (apolyakov) [reply tests/kphp_ci_pipeline_runner.py]: Fixed

**Review by astrophysik — APPROVED** [review]

File: tests/kphp_ci_pipeline_runner.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 import sys
 from enum import Enum
 
-from python.lib.colors import red, green, yellow, cyan
+from python.lib.colors import cyan, green, red, yellow
```
Comment (mkornaukhov03) [inline tests/kphp_ci_pipeline_runner.py:11]: Do we really need to break blame in such places?

**Review by mkornaukhov03 — APPROVED** [review]
Looks good
I wouldn't make formatting changes, but up to you
